editor/src/clj/editor/atlas.clj

```diff
@@ -1109,3 +1098,159 @@
                               node-child-index (.indexOf children node-id)]
                           (< node-child-index (dec (.size children)))))
   (run [selection] (move-node! (selection->image selection) 1)))
+
+(defn- snap-pivot
+  [pivot threshold]
+  (let [snap-values [0.0 0.5 1.0]]
+    (mapv #(or (some (fn [^double snap-value]
+                       (when (< (Math/abs (- % snap-value)) threshold)
+                         snap-value)) snap-values) %) pivot)))
+
+(defn- updated-pivot
+  [node-id ^Vector3d delta]
+  (let [scene (g/node-value node-id :scene)
+        rect (-> scene :renderable :user-data :rect)
+        {:keys [geometry width height]} rect
+        rotated (:rotated geometry)
+        [pivot-x pivot-y] (g/node-value node-id :pivot)
+        delta-x (/ (cond-> (.x delta) rotated -) width)
+        delta-y (/ (cond-> (.y delta) (not rotated) -) height)]
+    (if rotated
+      [(- pivot-x delta-y) (+ pivot-y delta-x)]
+      [(+ pivot-x delta-x) (+ pivot-y delta-y)])))
```

Comment:
I decided not to round here eventually. Since this is a normalized value, I am not sure about the maximum required precision.

Reply:
You should use either `properties/round-scalar` or `properties/round-scalar-float`, depending on if `pivot-x` is a `double` or a `float`. It's important to maintain the same numeric type as the value was before you transform it, otherwise it can mess with structural equality and hashing of before / after states. We have similar rounding functions for `coarse` values.

Have a look at `editor.scene/apply-move-delta`: https://github.com/defold/defold/blob/dev/editor/src/clj/editor/scene.clj#L1636-L1638